checks/elasticsearch.go

```diff
@@ -0,0 +1,96 @@
+package checks
+
+import (
+	//gocontext "context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/flanksource/canary-checker/api/context"
+	"github.com/flanksource/canary-checker/api/external"
+
+	v1 "github.com/flanksource/canary-checker/api/v1"
+	"github.com/flanksource/canary-checker/pkg"
+
+	"github.com/elastic/go-elasticsearch/v8"
+)
+
+type ElasticsearchChecker struct{}
+
+func (c *ElasticsearchChecker) Type() string {
+	return "elasticsearch"
+}
+
+func (c *ElasticsearchChecker) Run(ctx *context.Context) pkg.Results {
+	var results pkg.Results
+	for _, conf := range ctx.Canary.Spec.Elasticsearch {
+		results = append(results, c.Check(ctx, conf)...)
+	}
+	return results
+}
+
+func (c *ElasticsearchChecker) Check(ctx *context.Context, extConfig external.Check) pkg.Results {
+	check := extConfig.(v1.ElasticsearchCheck)
+	result := pkg.Success(check, ctx.Canary)
+	var results pkg.Results
+	results = append(results, result)
+
+	cfg := elasticsearch.Config{
+		Addresses: []string{check.GetEndpoint()},
+		Username:  check.Auth.GetUsername(),
+		Password:  check.Auth.GetPassword(),
+	}
+
+	es, err := elasticsearch.NewClient(cfg)
+	if err != nil {
+		return results.ErrorMessage(err)
+	}
+
+	body := strings.NewReader(check.Query)
+
+	res, err := es.Search(
+		// TODO: See if context is required here
+		//es.Search.WithContext(ctx),
+		es.Search.WithIndex(check.Index),
+		es.Search.WithBody(body),
+	)
+
+	if err != nil {
+		return results.ErrorMessage(err)
+	}
+
+	defer res.Body.Close()
```

Comment:
I would add this defer before the error check. Since in case of error we won't be closing the body since it is returning before adding this defer statement

Reply:
Makes sense, good catch :)